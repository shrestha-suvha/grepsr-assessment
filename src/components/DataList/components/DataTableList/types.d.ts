import { IDataSetList } from '../../types';
export interface IDataTableListProps {
  dataSetList: IDataSetList;
}
