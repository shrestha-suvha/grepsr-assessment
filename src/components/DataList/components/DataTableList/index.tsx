import React, { FC } from 'react';
import Table from 'react-bootstrap/Table';
import { FaHashtag } from 'react-icons/fa';
import { AiOutlineInfoCircle, AiOutlineFontColors } from 'react-icons/ai';
import { IDataTableListProps } from './types';

const DataTableList: FC<IDataTableListProps> = ({ dataSetList }) => {
  return (
    <div className="table-wrapper">
      <Table striped borderless className="fs-sm" style={{ maxHeight: '200px' }}>
        <thead>
          <tr>
            <th>
              <FaHashtag className="table__icon--hash " />
              <span className="fw-bold">geo_name</span>
              <AiOutlineInfoCircle className="table__icon--info" />
            </th>
            <th>
              <AiOutlineFontColors className="table__icon--hash " />
              <span className="fw-bold">location</span>
              <AiOutlineInfoCircle className="table__icon--info" />
            </th>
            <th>
              <FaHashtag className="table__icon--hash" />
              <span className="fw-bold">contact_no</span>
              <AiOutlineInfoCircle className="table__icon--info" />
            </th>
          </tr>
        </thead>
        <tbody className="fs-sm">
          {(dataSetList || []).map(dataSet => (
            <tr key={dataSet.id}>
              <td>{dataSet.geo_id}</td>
              <td>{dataSet.location}</td>
              <td>{dataSet.contact_no}</td>
            </tr>
          ))}
        </tbody>
      </Table>
    </div>
  );
};

export default DataTableList;
