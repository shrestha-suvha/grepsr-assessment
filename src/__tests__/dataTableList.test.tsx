import { render, screen, cleanup } from '@testing-library/react';
import { create } from 'react-test-renderer';

import DataTableList from 'src/components/DataList/components/DataTableList';

afterEach(() => {
  cleanup();
});

test('should render DataTableList component', () => {
  const dataSetList = [
    {
      geo_id: 'f9dd0ccdfc8e9ef1a7553c9a',
      location: 'Pontiac',
      contact_no: '(396) 401-6863 x0569',
      id: '1',
    },
  ];
  render(<DataTableList dataSetList={dataSetList} />);
  const dataSetListElement = screen.getByTestId('datasetlist-1');
  const tableElement = screen.getByRole('table');
  expect(dataSetListElement).toBeInTheDocument();
  expect(tableElement).toBeInTheDocument();
});

test('should render DataTableList with table header', () => {
  const dataSetList = [
    {
      geo_id: 'f9dd0ccdfc8e9ef1a7553c9a',
      location: 'Pontiac',
      contact_no: '(396) 401-6863 x0569',
      id: '1',
    },
  ];
  render(<DataTableList dataSetList={dataSetList} />);
  const tableHeaders = screen.getAllByRole('columnheader');
  expect(tableHeaders.length).toBe(3);
  expect(tableHeaders[0]).toHaveTextContent('geo_id');
  expect(tableHeaders[1]).toHaveTextContent('location');
  expect(tableHeaders[2]).toHaveTextContent('contact_no');
});

test('should render DataTableList component with props value', () => {
  const dataSetList = [
    {
      geo_id: 'f9dd0ccdfc8e9ef1a7553c9a',
      location: 'Pontiac',
      contact_no: '(396) 401-6863 x0569',
      id: '1',
    },
  ];
  render(<DataTableList dataSetList={dataSetList} />);
  expect(screen.getAllByRole('row')[1]).toHaveTextContent('f9dd0ccdfc8e9ef1a7553c9a');
  expect(screen.getAllByRole('row')[1]).toHaveTextContent('Pontiac');
  expect(screen.getAllByRole('row')[1]).toHaveTextContent('(396) 401-6863 x0569');
});

test('matches snapshot', () => {
  const tree = create(<DataTableList dataSetList={[]} />).toJSON();
  expect(tree).toMatchSnapshot();
});
