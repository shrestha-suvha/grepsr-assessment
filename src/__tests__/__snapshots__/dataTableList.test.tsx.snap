// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="table-wrapper"
  data-testid="datasetlist-1"
>
  <table
    className="fs-sm table table-striped table-borderless"
    style={
      Object {
        "maxHeight": "200px",
      }
    }
  >
    <thead>
      <tr>
        <th>
          <svg
            className="table__icon--hash "
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M440.667 182.109l7.143-40c1.313-7.355-4.342-14.109-11.813-14.109h-74.81l14.623-81.891C377.123 38.754 371.468 32 363.997 32h-40.632a12 12 0 0 0-11.813 9.891L296.175 128H197.54l14.623-81.891C213.477 38.754 207.822 32 200.35 32h-40.632a12 12 0 0 0-11.813 9.891L132.528 128H53.432a12 12 0 0 0-11.813 9.891l-7.143 40C33.163 185.246 38.818 192 46.289 192h74.81L98.242 320H19.146a12 12 0 0 0-11.813 9.891l-7.143 40C-1.123 377.246 4.532 384 12.003 384h74.81L72.19 465.891C70.877 473.246 76.532 480 84.003 480h40.632a12 12 0 0 0 11.813-9.891L151.826 384h98.634l-14.623 81.891C234.523 473.246 240.178 480 247.65 480h40.632a12 12 0 0 0 11.813-9.891L315.472 384h79.096a12 12 0 0 0 11.813-9.891l7.143-40c1.313-7.355-4.342-14.109-11.813-14.109h-74.81l22.857-128h79.096a12 12 0 0 0 11.813-9.891zM261.889 320h-98.634l22.857-128h98.634l-22.857 128z"
            />
          </svg>
          <span
            className="fw-bold"
          >
            geo_id
          </span>
          <svg
            className="table__icon--info"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
            />
            <path
              d="M464 336a48 48 0 1 0 96 0 48 48 0 1 0-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </th>
        <th>
          <svg
            className="table__icon--hash "
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M904 816H120c-4.4 0-8 3.6-8 8v80c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-80c0-4.4-3.6-8-8-8zm-650.3-80h85c4.2 0 8-2.7 9.3-6.8l53.7-166h219.2l53.2 166c1.3 4 5 6.8 9.3 6.8h89.1c1.1 0 2.2-.2 3.2-.5a9.7 9.7 0 0 0 6-12.4L573.6 118.6a9.9 9.9 0 0 0-9.2-6.6H462.1c-4.2 0-7.9 2.6-9.2 6.6L244.5 723.1c-.4 1-.5 2.1-.5 3.2-.1 5.3 4.3 9.7 9.7 9.7zm255.9-516.1h4.1l83.8 263.8H424.9l84.7-263.8z"
            />
          </svg>
          <span
            className="fw-bold"
          >
            location
          </span>
          <svg
            className="table__icon--info"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
            />
            <path
              d="M464 336a48 48 0 1 0 96 0 48 48 0 1 0-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </th>
        <th>
          <svg
            className="table__icon--hash"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M440.667 182.109l7.143-40c1.313-7.355-4.342-14.109-11.813-14.109h-74.81l14.623-81.891C377.123 38.754 371.468 32 363.997 32h-40.632a12 12 0 0 0-11.813 9.891L296.175 128H197.54l14.623-81.891C213.477 38.754 207.822 32 200.35 32h-40.632a12 12 0 0 0-11.813 9.891L132.528 128H53.432a12 12 0 0 0-11.813 9.891l-7.143 40C33.163 185.246 38.818 192 46.289 192h74.81L98.242 320H19.146a12 12 0 0 0-11.813 9.891l-7.143 40C-1.123 377.246 4.532 384 12.003 384h74.81L72.19 465.891C70.877 473.246 76.532 480 84.003 480h40.632a12 12 0 0 0 11.813-9.891L151.826 384h98.634l-14.623 81.891C234.523 473.246 240.178 480 247.65 480h40.632a12 12 0 0 0 11.813-9.891L315.472 384h79.096a12 12 0 0 0 11.813-9.891l7.143-40c1.313-7.355-4.342-14.109-11.813-14.109h-74.81l22.857-128h79.096a12 12 0 0 0 11.813-9.891zM261.889 320h-98.634l22.857-128h98.634l-22.857 128z"
            />
          </svg>
          <span
            className="fw-bold"
          >
            contact_no
          </span>
          <svg
            className="table__icon--info"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
            />
            <path
              d="M464 336a48 48 0 1 0 96 0 48 48 0 1 0-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </th>
      </tr>
    </thead>
    <tbody
      className="fs-sm"
    />
  </table>
</div>
`;
